File: vector/src/chatapp/java/com/chatapp/fragments/HomeModel.java
package com.chatapp.fragments;

import com.chatapp.C;

import java.util.ArrayList;

import im.vector.R;
import im.vector.VectorApp;

public class HomeModel {

    String name;
    String colorCode;
    int icon;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColorCode() {
        return colorCode;
    }

    public void setColorCode(String colorCode) {
        this.colorCode = colorCode;
    }

    public int getIcon() {
        return icon;
    }

    public void setIcon(int icon) {
        this.icon = icon;
    }

    public static ArrayList<HomeModel> getHomeList() {
        ArrayList<HomeModel> list = new ArrayList<>();
        Integer[] iconList =
                {
                        R.drawable.status_button,
                        R.drawable.invite_friends,
                        R.drawable.settings,
                        R.drawable.my_balance,
                        R.drawable.buy_credit,
                        R.drawable.voucher_recharge,
                        R.drawable.mobile_topup,
                        R.drawable.balance_transfer,
                        R.drawable.transfer_history,
//                        R.drawable.contacts_backup,
//                        R.drawable.tracking,
//                        R.drawable.did_icon,
                        R.drawable.meeting,
                        R.drawable.update_profile,
                        R.drawable.qr_code,
                        R.drawable.ticketing,
                        R.drawable.courier,
                        R.drawable.wills_smart_education,
                        R.drawable.wills_smart_medical,
                        R.drawable.law_enforcement,
                        R.drawable.agro_farming,
                        R.drawable.city_guide,
                        R.drawable.my_number,
                        R.drawable.why_wills,
                        R.drawable.logout
                };

        String[] colorList = {
                "#3398dc",
                "#39a8f0",
                "#268bcf",
                "#1c81c5",
                "#0b73ba",
                "#0064a8",
                "#0064a8",
                "#00579a",
                "#005095",
//                "#40c9c4",
//                "#10a19c",
                "#4c98cf",
                "#2481c5",
                "#4873a6",
                "#5A5485",
                "#524365",
                "#3398dc", "#39a8f0",
        };

        String[] titles = {
                C.Companion.getStatus(),
                C.Companion.getInviteFriends(),
                C.Companion.getSettings(),
                C.Companion.getMyBalance(),
                C.Companion.getBuyCredit(),
                C.Companion.getVoucherRecharge(),
                C.Companion.getMobileTopup(),
                C.Companion.getMobileTransfer(),
                C.Companion.getTrnasferHistory(),
                C.Companion.getMeeting(),
                C.Companion.getUpdateProfile(),
                C.Companion.getQr(),
                C.Companion.getTicketing(),
                C.Companion.getCourier(),
                C.Companion.getWillEducation(),
                C.Companion.getMedical(),
                C.Companion.getLaw(),
                C.Companion.getSmartAgro(),
                C.Companion.getSmartCityGuide(),
                C.Companion.getMyNumber(),
                C.Companion.getWhyWill(),
                C.Companion.getLogout()
        };
        for (int i = 0; i < iconList.length; i++) {
            HomeModel model = new HomeModel();
            model.colorCode = colorList[0];
            model.name = titles[i];
            model.icon = iconList[i];
            list.add(model);

        }
        return list;

    }

    public static ArrayList<HomeModel> getFirstHomeList() {
        ArrayList<HomeModel> list = new ArrayList<>();

        Integer[] iconList1 = {
                R.drawable.status,
                R.drawable.invite_friends,
                R.drawable.settings,
                R.drawable.mybalance,
                R.drawable.buy_credit,
                R.drawable.voucher,
                R.drawable.mobile_topup,
                R.drawable.transfer_credit,
                R.drawable.transfer_history,
        };
        Integer[] iconList = {
                R.drawable.status,
                R.drawable.invite_friends,
                R.drawable.my_number,
                R.drawable.settings,
                R.drawable.mybalance,
                R.drawable.buy_credit,
                R.drawable.voucher,
                R.drawable.mobile_topup,
                R.drawable.transfer_credit,
        };
        String[] colorList = {
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
        };

        String[] titles = {
                C.Companion.getStatus(),
                C.Companion.getInviteFriends(),
                C.Companion.getMyNumber(),
                C.Companion.getSettings(),
                C.Companion.getMyBalance(),
                C.Companion.getBuyCredit(),
                C.Companion.getVoucherRecharge(),
                C.Companion.getMobileTopup(),
                C.Companion.getMobileTransfer(),
//                C.Companion.getTOPB(),
                C.Companion.getInviteFriends(),
                C.Companion.getInviteFriends(),
                C.Companion.getInviteFriends(),
                C.Companion.getInviteFriends(),
//                C.Companion.getVT(),
                C.Companion.getInviteFriends(),
                C.Companion.getInviteFriends(),
                C.Companion.getInviteFriends()
        };
        for (int i = 0; i < iconList.length; i++) {
            HomeModel model = new HomeModel();
            model.colorCode = colorList[i];
            model.name = titles[i];
            model.icon = iconList[i];
            list.add(model);

        }
        return list;

    }

    public static ArrayList<HomeModel> getSecondHomeList() {
        ArrayList<HomeModel> list = new ArrayList<>();
        Integer[] iconList = {
                R.drawable.e_meeting,
                R.drawable.profile,
                R.drawable.qr,
                R.drawable.book_ticket,
                R.drawable.courier,
                R.drawable.education,
                R.drawable.medical,
                R.drawable.law,
                R.drawable.farming,
//                R.drawable.city_guide,
//                R.drawable.my_number,
//                R.drawable.why_wills,
//                R.drawable.logout
                };
        String[] colorList = {
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
        };

        String[] titles = {
                C.Companion.getMeeting(),
                C.Companion.getUpdateProfile(),
                C.Companion.getQr(),
                C.Companion.getTicketing(),
                C.Companion.getCourier(),
                C.Companion.getWillEducation(),
                C.Companion.getMedical(),
                C.Companion.getLaw(),
                C.Companion.getSmartAgro(),
//                C.Companion.getSmartCityGuide(),
//                C.Companion.getMyNumber(),
//                C.Companion.getWhyWill(),
//                C.Companion.getLogout()
        };
        for (int i = 0; i < iconList.length; i++) {
            HomeModel model = new HomeModel();
            model.colorCode = colorList[i];
            model.name = titles[i];
            model.icon = iconList[i];
            list.add(model);

        }
        return list;

    }
    public static ArrayList<HomeModel> getThirdHomeList() {
        ArrayList<HomeModel> list = new ArrayList<>();
        Integer[] iconList = {
//                R.drawable.meeting,
//                R.drawable.update_profile,
//                R.drawable.qr_code,
//                R.drawable.ticketing,
//                R.drawable.courier,
//                R.drawable.wills_smart_education,
//                R.drawable.wills_smart_medical,
//                R.drawable.law_enforcement,
//                R.drawable.agro_farming,
                R.drawable.city,
                R.drawable.my_number,
                R.drawable.why,
                R.drawable.logout};
        String[] colorList = {
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
        };

        String[] titles = {
//                C.Companion.getMeeting(),
//                C.Companion.getUpdateProfile(),
//                C.Companion.getQr(),
//                C.Companion.getTicketing(),
//                C.Companion.getCourier(),
//                C.Companion.getWillEducation(),
//                C.Companion.getMedical(),
//                C.Companion.getLaw(),
//                C.Companion.getSmartAgro(),
                C.Companion.getSmartCityGuide(),
                C.Companion.getMyNumber(),
                C.Companion.getWhyWill(),
                C.Companion.getLogout()
        };
        for (int i = 0; i < iconList.length; i++) {
            HomeModel model = new HomeModel();
            model.colorCode = colorList[i];
            model.name = titles[i];
            model.icon = iconList[i];
            list.add(model);

        }
        return list;

    }

//    Transfer History - web
//    Data Bundles Topup - web
//    Electricity Bills Payment - web
//    Television Bills Payment- web
//    Update profile - app
//    Why Wills Smart VoIP- app
    public static ArrayList<HomeModel> getSecHomeList() {
        ArrayList<HomeModel> list = new ArrayList<>();
        Integer[] iconList1 = {
                R.drawable.transfer_history,
                R.drawable.data_bundle,
                R.drawable.electricity,
                R.drawable.tv_recharge,
                R.drawable.transfer_funds,
                R.drawable.sms,
                R.drawable.my_number,
                R.drawable.profile,
                R.drawable.why};
        Integer[] iconList = {
                R.drawable.transfer_history,
                R.drawable.data_bundle,
                R.drawable.electricity,
                R.drawable.tv_recharge,
//                R.drawable.transfer_funds,
//                R.drawable.sms,
//                R.drawable.my_number,
                R.drawable.profile,
                R.drawable.why};
        String[] colorList = {
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.first_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.second_color),
                VectorApp.getInstance().getString(R.string.first_color),
        };

        String[] titles = {
                C.Companion.getTrnasferHistory(),
                C.Companion.getDataBundle(),
                C.Companion.getElectric(),
                C.Companion.getTV(),
//                C.Companion.getMONEYTRANSFER(),
//                C.Companion.getSMS(),
//                C.Companion.getMyNumber(),
                C.Companion.getUpdateProfile(),
                C.Companion.getWhyWill()
        };
        for (int i = 0; i < iconList.length; i++) {
            HomeModel model = new HomeModel();
            model.colorCode = colorList[i];
            model.name = titles[i];
            model.icon = iconList[i];
            list.add(model);

        }
        return list;

    }
}

File: vector/src/chatapp/java/com/chatapp/fragments/HomeModel.kt
package com.chatapp.fragments

import android.app.Activity
import com.chatapp.C
import com.chatapp.C.Companion.BuyCredit
import com.chatapp.C.Companion.CallDetailsReport
import com.chatapp.C.Companion.CallRates
import com.chatapp.C.Companion.ContactBackup
import com.chatapp.C.Companion.Courier
import com.chatapp.C.Companion.DataBundle
import com.chatapp.C.Companion.Did
import com.chatapp.C.Companion.ELearning
import com.chatapp.C.Companion.Electric
import com.chatapp.C.Companion.InviteFriends
import com.chatapp.C.Companion.Law
import com.chatapp.C.Companion.Logout
import com.chatapp.C.Companion.Medical
import com.chatapp.C.Companion.Meeting
import com.chatapp.C.Companion.MobileTopup
import com.chatapp.C.Companion.MobileTransfer
import com.chatapp.C.Companion.MyBalance
import com.chatapp.C.Companion.MyNumber
import com.chatapp.C.Companion.Qr
import com.chatapp.C.Companion.Settings
import com.chatapp.C.Companion.SmartCityGuide
import com.chatapp.C.Companion.Status
import com.chatapp.C.Companion.TV
import com.chatapp.C.Companion.Ticketing
import com.chatapp.C.Companion.TransferCash
import com.chatapp.C.Companion.TrnasferHistory
import com.chatapp.C.Companion.UpdateProfile
import com.chatapp.C.Companion.VoucherRecharge
import com.chatapp.C.Companion.WhyWill
import com.chatapp.C.Companion.WillEducation
import com.chatapp.C.Companion.bookId
import com.chatapp.C.Companion.callerId
import com.chatapp.C.Companion.services
import com.chatapp.C.Companion.smartAgro
import im.vector.R
import im.vector.VectorApp
import java.util.*

class HomeModel {
    var name:String = ""
    var colorCode = ""
    var icon = 0



    companion object {

        val firstHomeList: ArrayList<HomeModel>
            get() {
                val list = ArrayList<HomeModel>()
                val iconList = arrayOf(
                        R.drawable.status,
                        R.drawable.invite_friends,
                        R.drawable.caller_id,
                        R.drawable.profile,
                        R.drawable.mybalance,
                        R.drawable.did,
                        R.drawable.send_sms,
                        R.drawable.buy_credit,
                        R.drawable.voucher
                )
                val titles = arrayOf(
                        Status,
                        InviteFriends,
                        callerId,
                        UpdateProfile,
                        MyBalance,
                        Did,
                        C.SendSMS,
                        BuyCredit,
                        VoucherRecharge
                )
                for (i in iconList.indices) {
                    val model = HomeModel()
                    model.name = titles[i]
                    model.icon = iconList[i]
                    list.add(model)
                }
                return list
            }

        val thirdHomeList: ArrayList<HomeModel>
            get() {
                val list = ArrayList<HomeModel>()
                val iconList = arrayOf(
                        R.drawable.e_learning,
                        R.drawable.book_ticket,
                        R.drawable.courier,
                        R.drawable.contact_backup,
                        R.drawable.tracking,
                        R.drawable.teletalk_tv,
                        R.drawable.teletalk_store,
                        R.drawable.setting
                )
                val titles = arrayOf(
                        ELearning,
                        C.Ticketing,
                        Courier,
                        ContactBackup,
                        C.Tracking,
                        C.TeletakTV,
                        C.TeletalkStore,
                        C.Settings
                )
                for (i in iconList.indices) {
                    val model = HomeModel()
                    model.name = titles[i]
                    model.icon = iconList[i]
                    list.add(model)
                }
                return list
            }



        fun getSecHomeList(): ArrayList<HomeModel> {
            val list = ArrayList<HomeModel>()
            val iconList = arrayOf(
                    R.drawable.mobile_waltet,
                    R.drawable.transfer_credit,
                    R.drawable.transfer_history,
                    R.drawable.mobile_topup,
                    R.drawable.data_bundle,
                    R.drawable.electricity,
                    R.drawable.tv_recharge,
                    R.drawable.call_details_report,
                    R.drawable.call_rates
                    )
            val titles = arrayOf(
                    TransferCash,
                    MobileTransfer,
                    TrnasferHistory,
                    MobileTopup,
                    DataBundle,
                    Electric,
                    TV,
                    CallDetailsReport,
                    CallRates,
                    ELearning
            )
            for (i in iconList.indices) {
                val model = HomeModel()
                model.name = titles[i]
                model.icon = iconList[i]
                list.add(model)
            }
            return list
        }
    }
}